Invalid E0404 throw

This code:

```rust
let x: i8 += 14;
```

gives the following error:

```text
error[E0404]: expected trait, found builtin type `i8`
 --> src/main.rs:5:12
  |
5 |     let x: i8 += 14;
  |            ^^ not a trait

warning: trait objects without an explicit `dyn` are deprecated
 --> src/main.rs:5:12
  |
5 |     let x: i8 += 14;
  |            ^^^^ help: use `dyn`: `dyn i8 +`
  |
  = note: `#[warn(bare_trait_objects)]` on by default

error: aborting due to previous error
```

It's not clear or even close to what the actual problem is. I think it should throw the E0067 error code.

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->
This issue has been assigned to @mibac138 via [this comment](https://github.com/rust-lang/rust/issues/66736#issuecomment-624951014).
<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"mibac138"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/compiler/rustc_parse/src/parser/stmt.rs
+++ b/compiler/rustc_parse/src/parser/stmt.rs
@@ -291,7 +291,7 @@ impl<'a> Parser<'a> {
         Ok(P(ast::Local { ty, pat, init, id: DUMMY_NODE_ID, span: lo.to(hi), attrs, tokens: None }))
     }
 
-    /// Parses the RHS of a local variable declaration (e.g., '= 14;').
+    /// Parses the RHS of a local variable declaration (e.g., `= 14;`).
     fn parse_initializer(&mut self, eq_optional: bool) -> PResult<'a, Option<P<Expr>>> {
         let eq_consumed = match self.token.kind {
             token::BinOpEq(..) => {
@@ -306,6 +306,7 @@ impl<'a> Parser<'a> {
                     "=".to_string(),
                     Applicability::MaybeIncorrect,
                 )
+                .help("if you meant to overwrite, remove the `let` binding")
                 .emit();
                 self.bump();
                 true

--- a/src/test/ui/parser/let-binop.stderr
+++ b/src/test/ui/parser/let-binop.stderr
@@ -3,18 +3,24 @@ error: can't reassign to an uninitialized variable
    |
 LL |     let a: i8 *= 1;
    |               ^^ help: initialize the variable
+   |
+   = help: if you meant to overwrite, remove the `let` binding
 
 error: can't reassign to an uninitialized variable
   --> $DIR/let-binop.rs:6:11
    |
 LL |     let b += 1;
    |           ^^ help: initialize the variable
+   |
+   = help: if you meant to overwrite, remove the `let` binding
 
 error: can't reassign to an uninitialized variable
   --> $DIR/let-binop.rs:8:11
    |
 LL |     let c *= 1;
    |           ^^ help: initialize the variable
+   |
+   = help: if you meant to overwrite, remove the `let` binding
 
 error: aborting due to 3 previous errors
 

